Import the required libraries:
* `pandas`, for importing CSV files into a dataframe, and viceversa

In [1]:
import pandas as pd

Uncomment the following lines to download the data files from Zillow's webpage. Otherwise, the provided copies in `source_data` will be used.

In [2]:
# ZHVI for single family homes
# !wget -O source_data/Zip_Zhvi_SingleFamilyResidence.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_SingleFamilyResidence.csv
# ZHVI for multi-family homes
# !wget -O source_data/Zip_Zhvi_Condominum.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_Condominum.csv
# ZHVI for 1 bedroom houses
# !wget -O source_data/Zip_Zhvi_1bedroom.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_1bedroom.csv
# ZHVI for 2 bedroom houses
# !wget -O source_data/Zip_Zhvi_2bedroom.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_2bedroom.csv
# ZHVI for 3 bedroom houses
# !wget -O source_data/Zip_Zhvi_3bedroom.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_3bedroom.csv
# ZHVI for 4 bedroom houses
# !wget -O source_data/Zip_Zhvi_4bedroom.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_4bedroom.csv
# ZHVI for 5 or more bedroom houses
# !wget -O source_data/Zip_Zhvi_5BedroomOrMore.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zhvi_5BedroomOrMore.csv
# ZRI for all homes
# !wget -O source_data/Zip_Zri_AllHomesPlusMultifamily.csv http://files.zillowstatic.com/research/public/Zip/Zip_Zri_AllHomesPlusMultifamily.csv

Now we can import all the CSV files into dataframes

In [49]:
zhvi_sfh=pd.read_csv("source_data/Zip_Zhvi_SingleFamilyResidence.csv")
zhvi_sfh.shape
zhvi_mfh=pd.read_csv("source_data/Zip_Zhvi_Condominum.csv")
zhvi_mfh.shape
zhvi_1bd=pd.read_csv("source_data/Zip_Zhvi_1bedroom.csv")
zhvi_1bd.shape
zhvi_2bd=pd.read_csv("source_data/Zip_Zhvi_2bedroom.csv")
zhvi_2bd.shape
zhvi_3bd=pd.read_csv("source_data/Zip_Zhvi_3bedroom.csv")
zhvi_3bd.shape
zhvi_4bd=pd.read_csv("source_data/Zip_Zhvi_4bedroom.csv")
zhvi_4bd.shape
zhvi_5bd=pd.read_csv("source_data/Zip_Zhvi_5BedroomOrMore.csv")
zhvi_5bd.shape
zri_all=pd.read_csv("source_data/Zip_Zri_AllHomesPlusMultifamily.csv")
zri_all.shape

(13181, 120)

We also import the CSV file with the commute times generated earlier into another dataframe

In [76]:
commutetimes=pd.read_csv("prepared_data/zcta_commute_times_reduced.csv")
commutetimes.shape

(181, 10)

The next step for us is to remove unwanted columns from the dataframes with the housing indexes, as each ZHVI dataframe has 300 columns, and the ZRI dataframe has 120. In practice, we only need to keep the `RegionName` column, as that is the field that has to match the ZCTA, and the column with the index value for December 2019 (column `2019-12-31` in the ZHVI dataframes, and `2019-12` in the ZRI dataframes)

In [51]:
zhvi_sfh_thin=zhvi_sfh[['RegionName','2019-12-31']].copy()
zhvi_mfh_thin=zhvi_mfh[['RegionName','2019-12-31']].copy()
zhvi_1bd_thin=zhvi_1bd[['RegionName','2019-12-31']].copy()
zhvi_2bd_thin=zhvi_2bd[['RegionName','2019-12-31']].copy()
zhvi_3bd_thin=zhvi_3bd[['RegionName','2019-12-31']].copy()
zhvi_4bd_thin=zhvi_4bd[['RegionName','2019-12-31']].copy()
zhvi_5bd_thin=zhvi_5bd[['RegionName','2019-12-31']].copy()
zri_all_thin=zri_all[['RegionName', '2019-12']].copy()

As all these dataframes have the same column name, we are going to rename that column now to make iut easier to handle the merged dataframe later

In [52]:
zhvi_sfh_thin.rename(columns={"2019-12-31":"zhvi_sfh"},inplace=True)
zhvi_mfh_thin.rename(columns={"2019-12-31":"zhvi_mfh"},inplace=True)
zhvi_1bd_thin.rename(columns={"2019-12-31":"zhvi_1bd"},inplace=True)
zhvi_2bd_thin.rename(columns={"2019-12-31":"zhvi_2bd"},inplace=True)
zhvi_3bd_thin.rename(columns={"2019-12-31":"zhvi_3bd"},inplace=True)
zhvi_4bd_thin.rename(columns={"2019-12-31":"zhvi_4bd"},inplace=True)
zhvi_5bd_thin.rename(columns={"2019-12-31":"zhvi_5bd"},inplace=True)
zri_all_thin.rename(columns={"2019-12":"zri_all"},inplace=True)

The next step is to merge all the housing index columns from the housing dataframes with the commutetimes dataframe.

In [53]:
df_tmp = pd.merge(commutetimes, zhvi_sfh_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_mfh_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_1bd_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_2bd_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_3bd_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_4bd_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
df_tmp = pd.merge(df_tmp, zhvi_5bd_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del df_tmp["RegionName"]
commutetimes_housing = pd.merge(df_tmp, zri_all_thin, left_on="ZCTA5CE10", right_on="RegionName", how="inner").copy()
del commutetimes_housing["RegionName"]

In [65]:
commutetimes_housing.head(10)

STUSPS  ZCTA5CE10  ct_06:00  ct_06:30  ct_07:00  ct_07:30  ct_08:00  \
0     MD      20903   35.9334   43.6932   46.8932   50.1866   49.9400   
1     MD      20715   42.4998   48.8400   55.5534   57.2766   57.8668   
2     MD      20716   39.6434   47.4834   54.2600   56.0366   56.8734   
3     MD      20720   38.2502   46.2368   50.6466   53.3268   54.7566   
4     MD      20774   38.0966   46.4934   52.1066   55.9198   56.8066   
5     MD      20781   26.2168   32.0234   35.3202   38.0568   39.2702   
6     MD      20910   32.7868   36.4134   39.0332   42.0864   44.5034   
7     MD      20782   36.3498   41.0766   44.7736   48.1632   50.1766   
8     MD      20772   40.4268   45.3332   51.8900   52.4002   52.5000   
9     MD      20607   37.1334   41.2266   49.2400   50.9632   47.1234   

   ct_08:30  ct_09:00  ct_09:30  zhvi_sfh  zhvi_mfh  zhvi_1bd  zhvi_2bd  \
0   46.8766   41.2900   36.9100  411485.0  132811.0   97111.0  169494.0   
1   53.1632   46.0534   41.8268  353359.0  151136.0  184149.0  231795.0   
2   51.8068   44.6800   40.6600  332640.0  189470.0  138294.0  230586.0   
3   51.6768   43.6000   39.7900  432368.0  272576.0  398887.0  247074.0   
4   51.9134   44.3232   40.6934  367446.0  186530.0   97768.0  212789.0   
5   39.5266   34.7366   30.6000  362782.0  169546.0  128707.0  272726.0   
6   44.7200   44.0734   43.4834  635028.0  282197.0  257564.0  370160.0   
7   46.8368   41.3466   36.8364  357378.0   95165.0   82781.0  259208.0   
8   48.1964   43.5632   42.1234  347050.0  214246.0  135274.0  230998.0   
9   44.2568   38.6932   36.1466  408062.0  350248.0  377730.0  224939.0   

   zhvi_3bd  zhvi_4bd  zhvi_5bd  zri_all  
0  370863.0  443196.0  453451.0   2004.0  
1  344916.0  357194.0  427007.0   1895.0  
2  304802.0  364215.0  448082.0   1943.0  
3  338757.0  463750.0  503876.0   2080.0  
4  307462.0  394240.0  504181.0   1770.0  
5  398331.0  390723.0  366905.0   1617.0  
6  617643.0  664578.0  733117.0   1906.0  
7  357050.0  388460.0  391751.0   1849.0  
8  292993.0  383534.0  456070.0   1833.0  
9  309557.0  433167.0  466558.0      NaN

As we can see, there are some rows with `NaN` values, so we will get rid of them

In [67]:
commutetimes_housing=commutetimes_housing.dropna()
commutetimes_housing.shape

(114, 18)

As we can see, in `commutetimes_housing` we now have the dataframe with all the ZCTAs and their commutetimes and housing indexes. We should notice that we have gone down from 181 entries in the commute time CSV file to 114 now, as not all the ZCTAs have housing indexes available.

The final step for now is to save this dataframe. We will save 3 different CSV files to easy processing later if needed.
The first file will contain all the data in the dataframe:

In [71]:
commutetimes_housing.to_csv('prepared_data/zcta_commute_and_housing.csv', index=False)

The second file will contain only the housing information with the ZCTA:

In [72]:
commutetimes_housing[["ZCTA5CE10", "zhvi_sfh", "zhvi_mfh", "zhvi_1bd", "zhvi_2bd", "zhvi_3bd", "zhvi_4bd", "zhvi_5bd", "zri_all"]].to_csv('prepared_data/zcta_housing.csv', index=False)

The final file with contain only the commute time, but only for the rows that also have housing information available:

In [73]:
commutetimes_housing[["ZCTA5CE10", "ct_06:00", "ct_06:30", "ct_07:00", "ct_07:30", "ct_08:00", "ct_08:30", "ct_09:00", "ct_09:30"]].to_csv('prepared_data/zcta_commute_times_after_housing.csv', index=False)